File: src/main/java/com/github/zly2006/reden/access/ItemStackAccess.java
package com.github.zly2006.reden.access;

import com.github.zly2006.reden.itemShadow.ItemStackOwner;
import net.minecraft.item.ItemStack;

import java.util.List;

public interface ItemStackAccess {
    default void checkStackOwners() {
        getStackOwners().removeIf(it -> !it.checkContains((ItemStack) this));
    }

    List<ItemStackOwner> getStackOwners();
}

File: src/main/java/com/github/zly2006/reden/access/ItemStackAccess.kt
package com.github.zly2006.reden.access

import com.github.zly2006.reden.itemShadow.ItemStackOwner
import net.minecraft.item.ItemStack

@Suppress("INAPPLICABLE_JVM_NAME")
@JvmDefaultWithoutCompatibility
interface ItemStackAccess {
    fun checkStackOwners() {
        stackOwners.removeIf { this as ItemStack !in it }
    }

    @get:JvmName("getStackOwners\$reden")
    val stackOwners: MutableList<ItemStackOwner>
}
